spec/dependency_container_spec.rb

```diff
@@ -0,0 +1,82 @@
+require "spec_helper"
+
+describe Nurse::DependencyContainer do
+  let(:container) {
+    container = Nurse::DependencyContainer.new
+
+    container.define(Hash) do
+      Hash.new
+    end
+
+    container.define(:definition) do
+      :defined_object
+    end
+  }
+
+  describe "#define" do
+    it "defines dependencies" do
+      container.define(:foo) do
+        :bar
+      end
+
+      container.get(:foo).must_equal :bar
+    end
+  end
+
+  describe "#defined?" do
+    it "returns false when dependency was not defined" do
+      container.defined?(:foo).must_equal false
+    end
+
+    it "returns true when dependency was defined" do
+      container.defined?(Hash).must_equal true
+    end
+
+    it "handles classes as dependencies keys" do
+      container.defined?(Hash).must_equal true
+    end
+
+    it "handles symbols as dependencies keys" do
+      container.defined?(:definition).must_equal true
+    end
+
+    it "handles strings as dependencies keys" do
+      container.defined?('definition').must_equal true
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.